### DISCUSSION TITLE
[help] RMarkwodn report as tar_render() with dynamic tabs

### DISCUSSION THREAD

**Discussion by tkuentzler** [ropensci/tarchetypes#138] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

Hi!
I'm trying to create an rmarkdown-report, that is rendered in a targets-pipeline via tar_render(). The report is supposed to have a section with dynamically created tabs, that contain ggplots. The number of tabs/plots changes and is not known prior to running the pipeline. The rmarkdown-solution I found was to create a template.Rmd-file and use knitr:knit_expand() and knitr::knit(). The solution on its own works fine, which means I can knit the report manually. 

However, when I integrate the report into the targets-pipeline, I run into problems with the working directories. I have a structure, where the R-project folder (thus the beginning of here()) containts multiple sub-projects, each with a targets-pipeline in it. In yet another sub(sub)folder is the report. To better understand what happens, I turned keep_md: TRUE. I can see, that the plot-files are created in the directory of the report.rmd, however they are then searched for in the main here()-directory.

What works is storing the report in the main r-project-folder, where here() starts. This is not a solution I can really work with, so I am looking for a way to specify where the created plots for the tabs should be searched for. Or at least, that these documents are also created in the main here()-folder, since they are usually deleted after the report is created.

My guess is that this issue has something to do with tar_render()'s charactersitic "4. Forces the report to run in the user's current working directory instead of the working directory of the report" (from the help-file). Unfortunately, I have not idea how to change this behaviour. Can anyone help?

Since the folder-structure is difficult to describe I tried to create a reprex, let me know if it needs improvement. It's the first reprex I did in this way. https://github.com/tkuentzler/reprex_tarchetype_rmrarkdown

Thank you for any help in advance!

**Comment by wlandau** [comment, marked as answer]
I ran your reprex with `reprex_here/reprex_subproject` as the working directory, and unfortunately I could not reproduce the issue you describe. The output `report.html` successfully loads the plots (see below).

Have you tried https://github.com/ropensci/tarchetypes/discussions/134#discussioncomment-4719196 in the full project? This use case seems related to https://github.com/ropensci/tarchetypes/discussions/134.

![Screenshot 2023-02-21 at 3 51 39 PM](https://user-images.githubusercontent.com/1580860/220456055-24c2e128-40b8-4f0f-a0a2-ce6664ade2c7.png)

**Reply by tkuentzler** [reply]
Thank you very much for looking into it. I've checked and I see, why it worked for you. The folder ~output/report_files should not be there. This folder was created, when knitting manually, so when rendering via the pipeline, the plots are found. This was my mistake in setting up the reprex, because I was making sure, the rmarkdown itself was fine. When only using tar_render, without manually knitting beforehand, the report_files-folder does not exists (or when deleting the output/report_files-folder) and the problem should occure. I've deleted the folder from GitHub.

Prior to posting the issue, I've read the linked issue, but I could not understand how I could use knitr::include_graphics() dynamically, without knowing in advance the amount and names of the files to be used.

Thank you again, for looking into this.

**Reply by wlandau** [reply]
Thanks for fixing. I reproduced what I think you saw: the `report_files` folder with all the figures was written to `reprex_subproject` instead of `reprex_subproject/output`. I patched it in https://github.com/tkuentzler/reprex_tarchetype_rmrarkdown/pull/1. My recommendations are very similar to the ones from  https://github.com/ropensci/tarchetypes/discussions/134#discussioncomment-4719196: set the root directory as the pipeline root (the folder with the `_targets.R` file) and set other paths relative to the pipeline root. In your case, setting `fig.path` to `output/report_files/` worked. I would also recommend against `here()` unless it further helps enforce the `_targets.R` file as the marker of the pipeline root directory.

    ```{r setup, include = FALSE}
    root <- rprojroot::find_root_file(criterion = "_targets.R") # Find the pipeline root directory.
    knitr::opts_knit$set(root.dir = root) # Run the report from the pipeline root directory. 
    knitr::opts_chunk$set(fig.path = file.path(root, "output", "report_files/")) # Save figures to the correct place.
    ```

**Reply by tkuentzler** [reply]
Thank you very much, this fixed the issue! :)
